### PR TITLE
Improve computing validation metric in AutoMM distiller

### PR DESCRIPTION
**Pull request by zhiqiangdon**
1. Use `.update()` in computing the validation metric to compute `roc_auc` more accurately and avoid errors for `r2`.
2. Add one unit test for distillation.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/optimization/lit_distiller.py [hunk 1]
```diff
@@ -220,18 +220,20 @@ def _compute_loss(
...
         logits: torch.Tensor,
         label: torch.Tensor,
     ):
-        if isinstance(self.validation_metric, torchmetrics.AUROC):
+        if isinstance(metric, (torchmetrics.AUROC, torchmetrics.AveragePrecision)):
             prob = F.softmax(logits.float(), dim=1)
-            return self.validation_metric(preds=prob[:, 1], target=label)  # only for binary classification
-        elif isinstance(self.validation_metric, BaseAggregator):
-            return self.validation_metric(self.custom_metric_func(logits, label))
+            metric.update(preds=prob[:, 1], target=label)  # only for binary classification
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/optimization/lit_distiller.py:232]: Should we check the "full_state_update" flag in the metrics and decide how to trigger update?

See also https://torchmetrics.readthedocs.io/en/stable/pages/implement.html#internal-implementation-details .

Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/optimization/lit_distiller.py:232]: `full_state_update ` is used in `.forward()`. Since we only need `.update()`, we don't need it.

File: multimodal/src/autogluon/multimodal/optimization/lit_distiller.py [hunk 2]
```diff
@@ -291,12 +293,17 @@ def validation_step(self, batch, batch_idx):
...
+        ),
         self.log(
             self.validation_metric_name,
-            self._compute_metric(
-                logits=student_output[self.student_model.prefix][LOGITS],
-                label=batch[self.student_model.label_key],
-            ),
+            self.validation_metric,
+            on_step=False,
+            on_epoch=True,
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/optimization/lit_distiller.py:306]: I had the concern on whether we should reset the metrics but found this issue: https://github.com/Lightning-AI/lightning/issues/4396 . Thus, the current implementation is correct.

**Review by sxjscience — APPROVED** [review]
LGTM

**Comment by szha** [comment]
Job PR-1832-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1832/1/index.html

**Review by FANGAreNotGnu — APPROVED** [review]
